src/lib/pocketbase/Character/index.ts

```diff
@@ -23,31 +32,55 @@ export class CharacterFetcher extends PocketBase {
   }: {
     page: number;
     userId: Snowflake;
-  }): Promise<ListResult<CharacterData>> {
+  }): Promise<ListResult<Character>> {
     const response = await this.pb
       .collection(COLLECTIONS.characters)
-      .getList<CharacterData>(page, 10, {
+      .getList<Character>(page, 10, {
         filter: `userId="${userId}"`,
+        ...PocketBase.expand(...Object.values(RELATION_FIELD_NAMES)),
       });
 
     return response;
   }
 
-  public async getCharacterById(
-    id: CharacterData["id"]
-  ): Promise<CharacterData> {
+  // TODO: Update Validate Record so this code can be removed as getEntityById already does this
+  public async getCharacterById(id: Character["id"]): Promise<Character> {
     const response = await this.pb
       .collection(COLLECTIONS.characters)
-      .getOne<CharacterData>(id);
+      .getOne<Character>(id);
+
+    return response;
+  }
+
+  public async getEntityById<T extends AllowedEntityTypes>({
```

Comment:
This function is too generic to be inside a character fetcher. Character fetcher class should only work with the character collection